lib/class-wp-theme-json-resolver.php

```diff
@@ -26,6 +26,13 @@ class WP_Theme_JSON_Resolver {
 	 */
 	private static $theme = null;
 
+	/**
+	 * Whether or not the theme supports theme.json.
+	 *
+	 * @var boolean
+	 */
+	private static $theme_has_support = null;
```

Comment:
Just out of curiosity, is there a difference if we skip the `= null` part?
Usually I'd just write it like `private static $theme_has_support;`

Reply:
Nah, I don't think it'd matter in this particular case.

I understand it's a good practice to always initialize them and be explicit so you don't have to think about how the engine would do it and avoid weird cases:

- there's an uninitialized variable, you think the engine is going to initialized to null
- with that expectation, in certain places of your code you uses the `isset` check
- however, the engine decides to initialize it to an empty string because of some type checking it has done
- result: weird bug!

I don't know how much of this holds true for modern engines in any language VS being a convention people do to avoid weird scenarios.